server/src/controller/manifest.js

```diff
@@ -71,9 +71,10 @@ class ManifestController {
     return manifest.toJSON()
   }
 
-  // TODO: filters
-  async list () {
+  async list (name, sort, direction) {
     const manifests = await this.app.database.model.manifest
+      .where('name', 'LIKE', name + '%')
```

Comment:
Unsure as I'm not experienced with any techno used here, but the extra `%` makes me doubt: is this inserted as-is in the SQL query? In other words, is this field SQL-injectable using `name`?

Reply:
`Bookshelf` uses 'Knex' a query builder which uses query bindings to protect against sql injection by sending parameters and query string separately to database driver, as explained here:[https://stackoverflow.com/questions/49665023/does-knex-js-prevent-sql-injection](https://stackoverflow.com/questions/49665023/does-knex-js-prevent-sql-injection)

So the only way to enable sql injection is by writing
```javascript
knex.raw(`select * from foo where id = ${id}`) // NEVER DO THIS 
```
But the methods used here always uses bindings format, to see that we can set `debug: true` in the knex configuration under `server/src/database/bookshelf.js` and the query will be logged:

```javascript
{ method: 'select',
  options: {},
  timeout: false,
  cancelOnTimeout: false,
  bindings: [ 'thisIsATest%' ],
  __knexQueryUid: '6ab1e80b-7af1-457d-8b53-362f24b580df',
  sql: 'select "manifest".* from "manifest" where "name" like ? order by "manifest"."creation_date" desc' }

```